### ISSUE TITLE
Cubian x1 : Keyboard configuration

### ISSUE THREAD

**Issue by hipporigolo** [cubieplayer/Cubian#428]
Hello, 

I install a Cubian Desktop x1 on a Cubietruck. 
I met a problem when I tried to configure a french Keyboard (azerty). 
I follow the basic instruction : 

```
sudo apt-get install console-data
sudo dpkg-reconfigure keyboard-configuration
```

The modification are done, and it works, but after reboot, the qwerte keyboard come back. 
In de configuration file, the setting are still the right (PC 105 and fr), in the GUI keyboard configuration (menu system --> keyboard) the right keyboard is selected, but when I use the keyboard it is qwerty...
I had to reconfigure the keyboard each time... or a simplest way is to use the 

```
setxkbmap fr
```

after each reboot.
It is not really convenient.
==> Have you the same trouble, or do you succeed in configuring a alternative keyboard with Cubian X desktop ?
If yes, if you can give me your solution to change the keyboard, it will help me. (and I am sure that I miss something basic)

Thanks
Alex

**Comment by danfos** [comment]
If you can get want you want with:

```
setxkbmap fr
```

You can add it to your `~/.xinitrc` to make it permanent, see http://raspberrypi.stackexchange.com/questions/7990/how-can-i-change-the-keyboard-layout-for-the-gui-console-is-ok

**Comment by hipporigolo** [comment]
Hello, 

Thank you Danfos for your answer, but there is no .xinitrc in my ~. This file exists in Cubian with Mate ?

I try to add the setxkbmap fr is .bashrc and in .profile without success. 
Last weekend, I was with a friend and test the Cubietruck with his usb Keyboard (a DELL wired), and the keyboard was recongnised as an AZERTY keyboard as configured.
Go back at home, with my keyboard... is still QWERTY. So I think the problem come from my keyboard : it is a combo keyboard/mouse Logitech K520 (I also test a K360 with same result) ==> I read in thos forum that in past there was some trouble with logitech receiver solved by an option in kernel (HIDRAW replace hiddev ?). I read also something with udev rules 97-bluetooth-hid2hci.rules (but probably not for usb receiver ?)

So, to solve my problem, it seems there is 2 options : 
1) Be able to execute the command setxkbmap fr at the MATE startup ==> But I do not know the right file to add this command... if you can help me.
2) solved the problem of the logiteck wireless keyboard and HIDRAW/HIDDEV ... probably more complex and less accessible without a strong support from an expert.

Best whishes for the new year

Alex

**Comment by danfos** [comment]
@hipporigolo 

If you do not have a `~/.xinitrc` why not just create it.

**Comment by hipporigolo** [comment]
@danfos 

I tried by copiing the /etc/X11/xinit/xinitrc to ~/.xinitrc. I add setxkbmap fr on both file. 
But when I have the .xinitrc in my home, (for cubie user) I can not login, I have a message like "failed to execute login command"
So it seems that this file doesn't interact with MATE properly, perhaps there is an other way to launch a command at the MATE starting ?

Alex

**Comment by hipporigolo** [comment]
Hi, 

I improved the situation by adding a line in the slim.conf file:

```
# Commands executed when starting and exiting a session.
# They can be used for registering a X11 session with
# sessreg. You can use the %user variable
sessionstart_cmd        setxkbmap fr
# sessionstop_cmd       some command
```

With it, I have an AZERTY keyboard in the Session.
... but still a qwerty in SLIM for the login.
It seems difficult to solve the problem in slim, xorg does not manage correctly the unifying receiver of Logitech... It's a pitty, because this solution is very common and convenient with only 1 usb port used for both mouse and keyboard... and no wires.
I am still explore this problem with Google...

Alex

**Comment by hipporigolo** [comment]
I find these two post which could be a solution... but it is not really clear for me (explain for kernel expert?)
http://pu5h.info/article/xorg-ignores-keyboard-layout-with-logitech-unifying-receiver
https://bugs.freedesktop.org/show_bug.cgi?id=49950

If these posts could inspire someone.

Alex

**Comment by danfos** [comment]
If you have things for the session working by changing slim.conf you could try https://wiki.archlinux.org/index.php/SLiM#Change_Keyboard_Layout

**Comment by hipporigolo** [comment]
Yes, but it seems that xorg is not installed by default in cubian:
No folder xorg.conf.d in the folder X11
and when I try sudo apt-get install xorg, the system propose to install the new package. 
Currently I do not validate the xorg installation. 
Do you think I do? It will create more new trouble than it will resolve ?

in fact, I hesitate to use Slim with the bad keyboard and then prepare patching and compile the kernel... in several week or month (I hope a new Cubian version with a more recent kernel which solve the bug)

ALex
